include/glow/Importer/CommonOperatorLoader.h

```diff
@@ -705,6 +727,42 @@ class CommonOperatorLoader : public ProtobufLoader {
 
     return false;
   }
+
+  /// Utility function which computes the resulting shape in case of
+  /// multidirectional broadcasting.
+  std::vector<size_t>
+  computeMultidirectionalBroadcast(llvm::ArrayRef<size_t> shape0,
+                                   llvm::ArrayRef<size_t> shape1) {
+    size_t numDims0 = shape0.size();
+    size_t numDims1 = shape1.size();
+    size_t newNumDims = numDims0 > numDims1 ? numDims0 : numDims1;
+    std::vector<size_t> reshapeDims(newNumDims);
+
+    for (size_t i = 0; i < newNumDims; i++) {
+      reshapeDims[i] = 1;
+    }
+    mergeMultidirectionalBroadcast(reshapeDims, shape0);
+    mergeMultidirectionalBroadcast(reshapeDims, shape1);
+
+    return reshapeDims;
+  }
+
+private:
+  /// Merge shape \p shape into \p mergeShape, following multidirectional
+  /// broadcasting rules.
+  void mergeMultidirectionalBroadcast(std::vector<size_t> &mergeShape,
+                                      llvm::ArrayRef<size_t> shape) {
+    size_t shift = mergeShape.size() - shape.size();
+    for (size_t i = 0; i < shape.size(); i++) {
+      if (shape[i] != 1) {
+        assert(((shape[i] == mergeShape[shift + i]) ||
```

Comment:
I just merged #2079 which replaces all `assert`s in the model loading phase with llvm::Errors and this will reintroduce an `assert`, would you mind updating this to match the rest of the error handling in the loader

Reply:
yes, I've just noticed this merge :) I'm working on the change right now.

Reply:
For now, I fixed the merge conflicts. I still need to  change something for `assert`.

Reply:
@tlepley-cadence it looks great, thank you!